### DISCUSSION TITLE
File Push hangs and not work

### DISCUSSION THREAD

**Discussion by mauroreggio** [pglombardo/PasswordPusher#2383] [Q&A] [answered]
Hi
In my docker environment, version 1.41.15 before and 1.43.0 now (i update it today), i've problem pushing file. 
Is the first time i try to push a file, so i found the problem only today. The password push works good.
I not can see any usefull logs but the ui hangs in file upload at 100% and no go away (see image here follow).
The only log that i can see in the container, about this push process are:

`"PUT /rails/active_storage/disk/eyJfcmFpbHMiOnsiZGF0YSI6eyJrZXkiOiI4cHk2aXE5amdjaTZqczNvYmYzN2RjN2tlM2x6IiwiY29udGVudF90eXBlIjoiYXBwbGljYXRpb24vcGRmIiwiY29udGVudF9sZW5ndGgiOjI5Nzc3MzAsImNoZWNrc3VtIjoidFMU1d3RmcjBNdXRBR3JtZz09Iiwic2VydmljZV9uYW1lIjoibG9jYWwifSwiZXhwIjoiMjAyNC0wNy0zMVQwODoyNDoxMS4w==--1a53cc1cd08ed418214f3e86c HTTP/1.1" 413 183 "https://pwp.website.it/f/new" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:128.0) Gecko/20100101 Firefox/128.0" "-"`

`"GET /f/new HTTP/1.1" 200 11924 "https://pwp.website.it/f/new" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:128.0) Gecko/20100101 Firefox/128.0" "-"`

I think are nginx logs and after this, nothing.

I try search where file are stored and maybe check the folder permissions (linux host) ... but ... not know if i'm in the right directions.
Where the file are stored in docker env?
Have never see this?

Thanks.
Mauro.

![certificato](https://github.com/user-attachments/assets/7319e466-a6dd-4c0f-9911-435629e12195)

**Comment by pglombardo** [comment, marked as answer]
Hi Mauro,

Did you add a Docker volume to your container for local storage?  I have instructions [here](https://docs.pwpush.com/docs/self-hosted-configuration/#local-storage).

I believe that should be the solution.  Take a look at let me know.

**Reply by mauroreggio** [reply]
Nice ... is this .. sorry i missed it.
Thank you very much.
Mauro.

**Comment by ArxWachtMa** [comment]
Hello @pglombardo ,

Sorry for digging up this older thread, as the link doesn't work for me.

I am currently facing the same issue as OP, I recall using bind mounts in my older setups, I however am experiencing these hangs with or without:

Excerpt from docker compose:
Bind mount:
```
volumes:
 - type: bind
        source: /data/docker/pwpush_data/
        target: /opt/PasswordPusher/storage
   
```
No bind mount:
 /data/docker/pwpush_data/:/opt/PasswordPusher/storage

**Reply by pglombardo** [reply]
Thanks for pointing out the broken link.  I've updated it to: https://docs.pwpush.com/docs/self-hosted-configuration/#local-storage

Was this working for you before and stopped working recently?  Because I may have another theory if that is the case.

**Reply by ArxWachtMa** [reply]
Yes, this setup used to work for me fairly recently (last month). I do see that the volume is available in my container:
Docker inspect excerpt. 
```
"Mounts": [
            {
                "Type": "bind",
                "Source": "/data/docker/pwpush_data",
                "Destination": "/opt/PasswordPusher/storage",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            },
```

Version is pinned on 1.51.4 currently.  Please let me know if you'd like me to provide more comprehensive information.

**Reply by pglombardo** [reply]
Try the latest release.  V1.50.6.  There was a fix that may likely be related here.

I'm AFK but will be back soon.

**Reply by pglombardo** [reply]
Hi @ArxWachtMa - did you make any progress on this issue?

**Comment by ArxWachtMa** [comment]
Hey Peter,

I did. The container users uid and gid matched an unrelated user on my host system with insufficient write permissions.

I was able to resolve the issue by building from a local dockerfile where I just overwritten the args.

Sorry for the inconvenience.

Kind regards,
Maarten

Sent from Outlook for Android<https://aka.ms/AAb9ysg>a {
text-decoration: none;
}
a:visited {
color: #919395;
}
.jobtitle {
font-size:7.0pt;
font-family:Arial;
color: #919395;
}
.greytext {
font-size:9.0pt;
font-family:Arial;
color: #919395;
          text-decoration:none;
}
.blacktext {
font-size:9.0pt;
font-family:Arial;
color: black;
line-height:1.6
}







Maarten Wachters 
Cloud Custodian Engineer








 





tel.:       +32 3 450 67 90
email:   ***@***.***
www.arxus.eu  









Be green, keep it on the screen. Please think before you print this e-mail.
________________________________

**Reply by pglombardo** [reply]
Ok good to hear that it's resolved at least.  Let me know if anything else comes up!
